lib/manager/poetry/artifacts.ts

```diff
@@ -45,12 +72,18 @@ export async function updateArtifacts({
         cmd.push(`poetry update --lock --no-interaction ${dep}`);
       }
     }
+    const tagConstraint = getPythonConstraint(existingLockFileContent, config);
     const execOptions: ExecOptions = {
       cwdFile: packageFileName,
-      docker: { image: 'renovate/poetry' },
+      docker: {
+        image: 'renovate/python',
+        tagConstraint,
+        tagScheme: 'pep440',
+        preCommands: ['pip install poetry'],
```

Comment:
Is there any way we can work out which version of poetry is in use?

Presumably there's at least a way to tell the difference between 0.x and 1.x so if it's only a few lines of work we should maybe do that at least.

Reply:
Example: https://github.com/python-poetry/poetry/blob/9f93df1140efd44f37d07fc04fe5eb9abdc105db/pyproject.toml#L3

Reply:
I think we need to read the `build-system.requires` section to get `poetry` constraint.

https://github.com/renovate-tests/poetry1/blob/4f41e4287d81d9ba3688f8a9cf7d3f01bb08d25b/pyproject.toml#L19

https://python-poetry.org/docs/pyproject/#poetry-and-pep-517